vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/TestRunResultsListener.kt

```diff
@@ -156,7 +156,7 @@ class TestRunResultsListener(
     }
 
     private fun Map.Entry<Test, AndroidTestResult>.toTestResult(device: Device): TestResult {
-        val testInstanceFromBatch = testBatch.tests.find { "${it.pkg}.${it.clazz}" == key.clazz && it.method == key.method }
+        val testInstanceFromBatch = testBatch.tests.find { it == key }
```

Comment:
This was always null (checked with real run), so resultTest was taken from test result, not from the test batch - and as test result was created in DdmlibDeviceExtensions.toTest it doesn't have any annotation information - so it lost.